effectsize_supplement.Rmd

```diff
@@ -0,0 +1,281 @@
+import::from(MASS, mvrnorm)
+```
+
+```{r boilerplate, include = FALSE}
+format_num <- function(nums, sigdigits = 3) gsub("\\.$", "", formatC(nums, sigdigits, format = "fg", flag="#"))
+```
+
+
+## Data
+
+We will use the same data as the simple effect size exemplar: 
+
+```{r data_generation}
+set.seed(12)
+n <- 20
+data <- tibble(
+  group = rep(c("A", "B"), each = n),
+  completion_time_ms = c(
+    rlnorm(n, meanlog = log(170), sdlog = 0.3),
+    rlnorm(n, meanlog = log(50), sdlog = 0.4)
+  )
+)
+```
+
+See that exemplar for more information.
+
+
+## Calculating simple effect size
+
+### Approach 1: Difference in means with Student's t confidence interval
+
+This is the approach used in the exemplar. While the response distributions are non-normal, the sampling distribution of the difference in means will still be defined on $(-\infty, +\infty)$ and approximately symmetrical [citation needed / double check this], so we can compute a *Student's t distribution confidence interval* for the difference in means.
+
+```{r t_test}
+t_result <- 
+  t.test(completion_time_ms ~ group, data = data) %>%
+  tidy()    # put result in tidy tabular format
+t_result
+```
+
+The `tidy()`ed output of the `t.test()` function includes an estimate of the mean difference in milliseconds (`estimate`) as well as the lower (`conf.low`) and upper (`conf.high`) bounds of the 95% confidence interval. 
+
+### Approach 2a: Ratio of medians with Student's t confidence interval on log-scale
+
+For responses that are assumed to be log-normal, one alternative is to calculate the mean difference on the log scale. Because the mean on the log scale of a log-normal distribution is equal to the median of the distribution of the untransformed responses, this is equivalent to calculating a ratio of medians on the untransformed scale (in this case, a ratio of median response times). See the [data transformation](data_transformation) guideline for more information.
+
+```{r log_t_test}
+log_t_result <- 
+  t.test(log(completion_time_ms) ~ group, data = data) %>%
+  tidy()    # put result in tidy tabular format
+log_t_result
+```
+
+We can transform this into a ratio of median response times:
+
+```{r log_t_to_ratio}
+log_t_ratios <- log_t_result %>%
+  mutate_at(vars(estimate, estimate1, estimate2, conf.low, conf.high), exp)
+log_t_ratios
+```
+
+This output shows the estimated median response times (`estimate1` and `estimate2`), and an estimate of the ratio between them (`estimate = estimate1/estimate2`) as well as the lower (`conf.low`) and upper (`conf.high`) bounds of the 95% confidence interval of that ratio. This allows us to estimate how many times slower or faster a given condition is. 
+
+However, since we have some sense in this context of how large or small we might want response times to be on the original scale (e.g., people tend to perceive differences on the order of 100ms), it may be easier to interpret effect sizes if we calculate them on that scale.
```

Comment:
May be we should show the calculation of the ratio back to the ms to clarify this? For example:

> The geometric mean of condition A is 153.92 and of B is 51.99. 
> A is roughly 2.96 times to B with the 95% confidence interval of [2.43, 3.61] times.
> 51.99 * 2.96 ≈153.92